Fix/test build failure

buschbapti opened the pull request:
`state_representation` and `controllers` where failing to build due to last minute changes. This corrects both issues.

eeberhard reviewed (approved):
Great, this fixes the compilation issues! The DS test still fails but I understand that was not the scope of this PR. Therefore we now see the docker build fail at the `make test` stage, but the `make` stage succeeds.

buschbapti commented:
`TestLinear` is now corrected. The problem was exactly what you mentioned earlier regarding the operators. The operation to integrate the twist in the next position was:

```
current_pose = dt * twist + current_pose;
```

As you figured out, + operator is not commutative, hence the correct way is:

```
current_pose = current_pose + dt * twist;
```

or 

```
current_pose += dt * twist;
```

this is indeed problematic. It needs a more thorough thinking. At least the test is now corrected. I have also cleaned up a bit using UnitRandom quaternions.

domire8 reviewed (approved):
Very nice! build-test.sh is now successful!

eeberhard commented:
> `TestLinear` is now corrected. The problem was exactly what you mentioned earlier regarding the operators. The operation to integrate the twist in the next position was:
> 
> ```
> current_pose = dt * twist + current_pose;
> ```
> 
> As you figured out, + operator is not commutative, hence the correct way is:
> 
> ```
> current_pose = current_pose + dt * twist;
> ```
> 
> or
> 
> ```
> current_pose += dt * twist;
> ```
> 
> this is indeed problematic. It needs a more thorough thinking. At least the test is now corrected. I have also cleaned up a bit using UnitRandom quaternions.

could you please leave in the opposite case and EXPECT_NE (or equivalent)? Just so it's super clear that the commutation matters in this case?

Or maybe that test belongs in the state operation itself.
